src/wallet/asyncrpcoperation_sendmany.h

```diff
@@ -0,0 +1,169 @@
+// Copyright (c) 2016 The Zcash developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef ASYNCRPCOPERATION_SENDMANY_H
+#define ASYNCRPCOPERATION_SENDMANY_H
+
+#include "asyncrpcoperation.h"
+#include "amount.h"
+#include "base58.h"
+#include "primitives/transaction.h"
+#include "zcash/JoinSplit.hpp"
+#include "zcash/Address.hpp"
+#include "json/json_spirit_value.h"
+#include "wallet.h"
+
+#include <tuple>
+
+// TODO: Compute fee based on a heuristic, e.g. (num tx output * dust threshold) + joinsplit bytes * ?
+#define ASYNC_RPC_OPERATION_DEFAULT_MINERS_FEE   10000
+
+using namespace libzcash;
+using namespace json_spirit;
+
+// A recipient is a tuple of address, amount, memo (optional if zaddr)
+typedef std::tuple<std::string, CAmount, std::string> SendManyRecipient;
```

Comment:
File a ticket to change these to structs / plain data classes, to avoid having to refer to fields by number.